lectures/wealth_dynamics.md

```diff
@@ -35,600 +49,752 @@ Let's check the GPU we are running
+    y_i =
+       \frac{\sum_{j \leq i} w_j}{\sum_{j \leq n} w_j}  
+\end{equation*}
 
 ```{code-cell} ipython3
-def generate_lorenz_and_gini_jax(wdy, num_households=100_000, T=500):
-    """
-    Generate the Lorenz curve data and gini coefficient corresponding to a
-    WealthDynamics mode by simulating num_households forward to time T.
-    """
-    size = (num_households, )
-    ψ_0 = jnp.full(size, wdy.y_mean)
-    ψ_star = update_cross_section_jax(ψ_0, T, wdy, size)
-    return gini_jax(ψ_star), lorenz_curve_jax(ψ_star)
+def _lorenz_curve_jax(w, w_size):
+    n = w.shape[0]
+    w = jnp.sort(w)
+    x = jnp.arange(n + 1) / n
+    s = jnp.concatenate((jnp.zeros(1), jnp.cumsum(w)))
+    y = s / s[n]
+    return x, y
 
-# Create the jit function
-generate_lorenz_and_gini_jax = jax.jit(generate_lorenz_and_gini_jax,
-                                       static_argnums=(1,2,))
+lorenz_curve_jax = jax.jit(_lorenz_curve_jax, static_argnums=(1,))
 ```
 
-Now we investigate how the Lorenz curves associated with the wealth distribution change as return to savings varies.
-
-The code below plots Lorenz curves for three different values of $\mu_r$.
+Let's test
 
 ```{code-cell} ipython3
-%%time
-
-fig, ax = plt.subplots()
-μ_r_vals = (0.0, 0.025, 0.05)
-gini_vals = []
-
-for μ_r in μ_r_vals:
-    wdy = create_wealth_model(μ_r=μ_r)
-    gv, (f_vals, l_vals) = generate_lorenz_and_gini_jax(wdy)
-    ax.plot(f_vals, l_vals, label=f'$\psi^*$ at $\mu_r = {μ_r:0.2}$')
-    gini_vals.append(gv)
+sim_length = 200
+num_households = 1_000_000
+ψ_0 = jnp.full(num_households, y_mean)  # Initial distribution
+z_sequence = generate_aggregate_state_sequence(aggregate_params,
+                                               length=sim_length)
+z_sequence = jnp.array(z_sequence)
+```
 
-ax.plot(f_vals, f_vals, label='equality')
-ax.legend(loc="upper left")
-plt.show()
+```{code-cell} ipython3
+key = jax.random.PRNGKey(1234)
+ψ_star = update_cross_section_jax_compiled(
+        model, ψ_0, num_households, z_sequence, key
+)
 ```
 
-The Lorenz curve shifts downwards as returns on financial income rise, indicating a rise in inequality.
+```{code-cell} ipython3
+%time _ = lorenz_curve_jax(ψ_star, num_households)
+```
 
-Now let's check the Gini coefficient.
+```{code-cell} ipython3
+%time x, y = lorenz_curve_jax(ψ_star, num_households)
+```
 
 ```{code-cell} ipython3
 fig, ax = plt.subplots()
-ax.plot(μ_r_vals, gini_vals, label='gini coefficient')
-ax.set_xlabel("$\mu_r$")
+ax.plot(x, y, label="Lorenz curve at defaults")
+ax.plot(x, x, 'k-', lw=1)
 ax.legend()
 plt.show()
 ```
 
-Once again, we see that inequality increases as returns on financial income
-rise.
+#### Gini Coefficient
 
-Let's finish this section by investigating what happens when we change the
-volatility term $\sigma_r$ in financial returns.
+Recall that, for sorted data $w_1, \ldots, w_n$, the Gini coefficient takes the form
 
-```{code-cell} ipython3
-%%time
+\begin{equation}
+    \label{eq:gini}
```

Comment:
@HumphreyYang I don't think that this label will get parsed. 

It would be better to wrap the latex in `$$` or `math` directive with a label `()` attribute